tests/pkg_semtech-loramac/main.c

```diff
@@ -76,49 +74,49 @@ static int _cmd_loramac(int argc, char **argv)
 
         if (strcmp("deveui", argv[2]) == 0) {
             uint8_t deveui[LORAMAC_DEVEUI_LEN];
-            semtech_loramac_get_deveui(deveui);
+            semtech_loramac_get_deveui(&loramac, deveui);
```

Comment:
I would have preferred here the usage of IPC rather than direct calls + mutex. IMO:
- If there's a single entry point for all commands, there shouldn't be needs for mutex (e.g [here](https://github.com/RIOT-OS/RIOT/blob/master/pkg/openthread/contrib/netdev/openthread_netdev.c#L122))
- In can be integrated to netapi, since most of these commands are network options.

What do you think?

Reply:
Basically, this means reusing MSG_TYPE_LORAMAC_CMD

Reply:
I understand what you mean and I avoided it on purpose. That would have required other internal get/set functions, with switch/case depending on the parameters, very similar to netdev.

The actual approach is simpler and I think it ends up to the same result.

If the semtech-loramac API is later integrated into netif, then it will be possible to call it in the netapi. I need to rework [my first attempt](https://github.com/aabadie/RIOT/tree/pkg_semtech-loramac-netif) to achieve this.